src/charm.py

```diff
@@ -88,22 +83,14 @@ def __init__(self, *args):
         self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.on.restart_action, self._on_restart_action)
 
-        # Handle db:pgsql and visibility:pgsql relations. The "db" and
-        # "visibility" strings in this code block reflect the relation names.
+        # Handle postgresql relation.
         self.db = DatabaseRequires(self, relation_name="db", database_name=DB_NAME)
-        self.framework.observe(self.db.on.database_created, self._on_database_changed)
-        self.framework.observe(self.db.on.endpoints_changed, self._on_database_changed)
-        self.framework.observe(self.on.db_relation_broken, self._on_database_relation_broken)
-
         self.visibility = DatabaseRequires(self, relation_name="visibility", database_name=VISIBILITY_DB_NAME)
-        self.framework.observe(self.visibility.on.database_created, self._on_database_changed)
-        self.framework.observe(self.visibility.on.endpoints_changed, self._on_database_changed)
-        self.framework.observe(self.on.visibility_relation_broken, self._on_database_relation_broken)
+        self.postgres_actions = Postgresql(self)
```

Comment:
why is this called `_actions` but the others aren't?

Reply:
No reason in specific, will merge for now and remove ```_actions``` in the next PR.